x/pairing/keeper/msg_server_stake_provider_test.go

```diff
@@ -847,19 +849,291 @@ func TestCommisionChange(t *testing.T) {
+		{"stake with vault", p1Acc.Vault.Addr, p1Acc.Addr, ts.spec, true},
+		{"stake with operator", p1Acc.Addr, p1Acc.Addr, ts.spec, false},
+	}
+
+	for i, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			beforeVault := ts.GetBalance(tt.vault)
+			beforeOperator := ts.GetBalance(tt.operator)
+			err := ts.StakeProviderExtra(tt.operator.String(), tt.vault.String(), tt.spec, testStake+(100*int64(i+1)), []epochstoragetypes.Endpoint{{Geolocation: 1}}, 1, "test")
+			afterVault := ts.GetBalance(tt.vault)
+			afterOperator := ts.GetBalance(tt.operator)
+
+			ts.AdvanceEpoch()
+
+			if !tt.valid {
+				require.Error(t, err)
+
+				// balance
+				require.Equal(t, beforeVault, afterVault)
+				require.Equal(t, beforeOperator, afterOperator)
+
+				// stake entry
+				_, found := ts.Keepers.Epochstorage.GetStakeEntryByAddressCurrent(ts.Ctx, tt.spec.Index, tt.operator.String())
+				require.True(t, found) // should be found because operator is registered in a vaild stake entry
+
+				// delegations
+				res, err := ts.QueryDualstakingDelegatorProviders(tt.vault.String(), false)
+				require.NoError(t, err)
+				require.Len(t, res.Delegations, 0)
+			} else {
+				require.NoError(t, err)
+
+				// balance
+				require.Equal(t, beforeVault-100, afterVault)
+				if !tt.operator.Equals(tt.vault) {
+					require.Equal(t, beforeOperator, afterOperator)
+				} else {
+					require.Equal(t, beforeOperator-100, afterOperator)
+				}
+
+				// stake entry
+				stakeEntry, found := ts.Keepers.Epochstorage.GetStakeEntryByAddressCurrent(ts.Ctx, tt.spec.Index, tt.operator.String())
+				require.True(t, found)
+				require.Equal(t, testStake+100, stakeEntry.Stake.Amount.Int64())
+				require.Equal(t, int64(0), stakeEntry.DelegateTotal.Amount.Int64())
+
+				// delegations
+				res, err := ts.QueryDualstakingDelegatorProviders(tt.vault.String(), false)
+				require.NoError(t, err)
+				require.Len(t, res.Delegations, 1)
+				require.Equal(t, tt.operator.String(), res.Delegations[0].Provider)
+				require.Equal(t, testStake+100, res.Delegations[0].Amount.Amount.Int64())
+			}
+		})
+	}
+}
+
+// TestVaultOperatorModifyStakeEntry tests that the operator can change non-stake related properties of the stake entry
+// and the vault address can change anything
+// Scenarios:
+// 1. operator modifies all non-stake related properties -> should work
+// 2. operator modifies all stake related properties -> should fail
+// 3. vault can change anything in the stake entry
+func TestVaultOperatorModifyStakeEntry(t *testing.T) {
+	ts := newTester(t)
+	ts.setupForPayments(1, 1, 0)
+	acc, _ := ts.GetAccount(common.PROVIDER, 0)
+	_, dummy := ts.GetAccount(common.CONSUMER, 0)
+
+	operator := acc.Addr.String()
+	vault := acc.Vault.Addr.String()
+	valAcc, _ := ts.GetAccount(common.VALIDATOR, 0)
+
+	stakeEntry, found := ts.Keepers.Epochstorage.GetStakeEntryByAddressCurrent(ts.Ctx, ts.spec.Index, acc.Addr.String())
+	require.True(t, found)
+
+	// consts for stake entry changes
+	var (
```

Comment:
Then why not const?

Reply:
Then you can use `iota`

Reply:
done. see https://github.com/lavanet/lava/commit/88b007bc6f553ad6d48b971c823b1f0f4cb3738b